reposerver/repository/repository.go

```diff
@@ -2573,6 +2573,25 @@ func (s *Service) GetGitFiles(_ context.Context, request *apiclient.GitFilesRequ
 		return nil, status.Errorf(codes.Internal, "unable to resolve git revision %s: %v", revision, err)
 	}
 
+	if gpg.IsGPGEnabled() && request.VerifyCommit {
```

Comment:
i would put it in private function, like: verifyCommitSignature, which will just return error